addons/binding/org.openhab.binding.mysensors/src/main/java/org/openhab/binding/mysensors/internal/factory/MySensorsCacheFactory.java

```diff
@@ -0,0 +1,138 @@
+/**
+ * Copyright (c) 2014-2016 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.mysensors.internal.factory;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.reflect.Type;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * ID Cache is used to store the ids used in the MySensors network.
+ *
+ * @author Andrea Cioni
+ *
+ */
+public class MySensorsCacheFactory {
+    private Logger logger = LoggerFactory.getLogger(getClass());
+
+    private static MySensorsCacheFactory singleton = null;
+
+    private static final String CACHE_BASE_PATH = "./mysensors/cache";
```

Comment:
Where would this cache be written in a production environment? I believe it should belong into userdata, but not into the current working directory of the process. To which you might even not have write access....